src/witan/send/adroddiad/clerk/charting_v2.clj

```diff
@@ -145,3 +152,611 @@
+                                 :legend {:gradientLength 200}
+                                 :scale {:domainMid 0
+                                         :scheme "blueorange"}
+                                 :title color-field
+                                 :type "quantitative"}
+                         :tooltip tooltip}
+              :mark "rect"}
+             {:encoding {:color {:condition {:test white-text-test
+                                             :value "white"}
+                                 :value "black"}
+                         :text {:field color-field :type "quantitative"}
+                         :tooltip tooltip}
+              :mark "text"}]}))
+
+(defn ehcp-heatmap-per-year
+  [census x-field]
+  (clerk/vl
+   {::clerk/width :full}
+   (let [color-field     "Row Count"
+         x-field         x-field
+         x-order-field   (sort-field x-field)
+         x-label         (field-descriptions x-field)
+         y-field         :calendar-year
+         data            (-> census
+                             (tc/group-by [x-field y-field])
+                             (tc/aggregate {color-field tc/row-count})
+                             (tc/complete x-field y-field)
+                             (tc/replace-missing color-field :value 0)
+                             (tc/map-columns x-label [x-field]
+                                             (fn [s] s))
+                             (tc/order-by [x-label :calendar-year])
+                             (tc/add-column :order (range)))
+         white-text-test (format "datum['%s'] > %d"
+                                 (name color-field)
+                                 (int (+ (reduce dfn/min (data color-field))
+                                         (* 0.450 (- (reduce dfn/max (data color-field))
+                                                     (reduce dfn/min (data color-field)))))))
+         ]
+     (heatmap-desc
+      {:data            (-> data
+                            (tc/rows :as-maps))
+       :height          150
+       :width           (cond
+                          (= x-field :academic-year)
+                          full-width
+                          (= x-field :scenario)
+                          two-rows
+                          :else
+                          half-width)
+       :y-field         (sweet-column-names y-field y-field)
+       :y-field-desc    (field-descriptions y-field y-field)
+       :y-field-label   (axis-labels y-field)
+       :x-field         (sweet-column-names x-field x-field)
+       :x-field-desc    (field-descriptions x-field x-field)
+       :x-field-label   (axis-labels x-field)
+       :x-sort-field    x-order-field
+       :color-field     color-field
+       :title           "# EHCPs for Settings by Year"
+       :white-text-test white-text-test
+       }))))
+
+(defn ehcps-total-by-year
+  [census]
+  (-> census
+      (tc/add-column :scenario "Baseline")
```

Comment:
I think this might need a configuration map passed in where you can say what the scenario is, but it can default via an `:or` destructuring to `Baseline`